### ISSUE TITLE
MaskedInputFormatter new allowedCharMatcher parameter appears to apply to numbers as well as characters

### ISSUE THREAD

**Issue by michael-woolsey** [caseyryan/flutter_multi_formatter#60]
Hello,

On version `flutter_multi_formatter: ^2.2.0` I was using `MaskedInputFormatter('#0#0#0', anyCharMatcher: RegExp('[a-z]', caseSensitive: false))` where the parameter was `anyCharMatcher` and I was getting my formatting working as expected where I could have `h0h0h0`.

However when I updated to `flutter_multi_formatter: ^2.3.5` I updated the method to the new `allowedCharMatcher` like this `MaskedInputFormatter('#0#0#0', allowedCharMatcher: RegExp('[a-z]', caseSensitive: false))` and now I cannot enter my numbers.

It seems like the `allowedCharMatcher` is applying to every item in the string instead of just the # which I believe is what it's supposed to do as stated in the comments:
```
  /// will match literally any character unless
  /// you supply an [allowedCharMatcher] parameter with a RegExp
  /// to constrain its values. e.g. RegExp(r'[a-z]+') will make #
  /// match only lowercase latin characters and everything else will be
  /// ignored
```

Did I do something wrong?

**Comment by doitduri** [comment]
Hi, 
I'm having the same problem.
You entered the regular expression correctly, but it is not entered. Please check.

```
inputFormatters: [
        MaskedInputFormatter('#0#0#0', allowedCharMatcher: RegExp(r'([a-z][0-9])\3*'),)
      ],
```
![image](https://user-images.githubusercontent.com/26545623/132543268-788d93a4-9b45-481e-9423-c34b4dc99327.png)

**Comment by michael-woolsey** [comment]
@doitduri - I tried your regular expression and unfortunately it does not work.

When I change it to `MaskedInputFormatter('#0#0#0', allowedCharMatcher: RegExp(r'([a-z][0-9])\3*'),)` I am no longer allowed to enter anything, no characters, no numbers, nothing.  I tried with and without `caseSensitive` and in both cases it still doesn't allow me to enter anything.

**Comment by caseyryan** [comment]
Hi! 
I'll take a look at it as soon as I have time. Hopefully this weekend. 
Thanks for reporting

**Comment by caseyryan** [comment]
Fixed in version 2.3.7
